tools/ci_build/github/azure-pipelines/c-api-packaging-pipelines.yml

```diff
@@ -54,7 +54,7 @@ jobs:
     vmImage: 'macOS-10.13'
   steps:
   
-    - template: templates/set-test-data-variables-step.yml
+    - template: templates/mac-set-variables-and-download.yml
     - template: templates/set-version-number-variables-step.yml
```

Comment:
The Windows jobs can be shortened/refactored in general.

For example the **Windows_Packaging_GPU_x64** (lines 138-175), can be reduced more than 50% by doing something similar to  **win-gpu-ci-pipeline.yml**. Not critical, but would be good to reduce the number of lines of this script in general.

Reply:
Hi @jignparm,  Your suggestion is good, but I'd prefer to keep this PR small and defer the change to another pr.  So that if something went wrong after the checkin, it would be easier to debug and fix the issue. Thank you.